core/dbt/clients/jinja.py

```diff
@@ -424,3 +468,38 @@ def extract_toplevel_blocks(
         allowed_blocks=allowed_blocks,
         collect_raw_data=collect_raw_data
     )
+
+
+SCHEMA_TEST_KWARGS_NAME = '_dbt_schema_test_kwargs'
+
+
+def add_rendered_test_kwargs(
+    context: Dict[str, Any],
+    node: Union[ParsedSchemaTestNode, CompiledSchemaTestNode],
+    capture_macros: bool = False,
+) -> None:
+    """Render each of the test kwargs in the given context using the native
+    renderer, then insert that value into the given context as the special test
+    keyword arguments member.
+    """
+    looks_like_func = r'^\s*(env_var|ref|var|source|doc)\s*\(.+\)\s*$'
+
+    # we never care about the keypath
+    def _convert_function(value: Any, _: Any) -> Any:
+        if isinstance(value, str):
+            if re.match(looks_like_func, value) is not None:
+                # curly braces to make rendering happy
+                value = f'{{{{ {value} }}}}'
```

Comment:
Do we have the ability to reach into the `context` and call these functions directly? I think the approach shown here is a-ok for now, but I'd like to remove this jinja hackery in the future if we can!

Reply:
we could, but it would be quite tricky to handle things like `ref(var('foo'))` because you have to recursively descend into the arguments

Reply:
oh, interesting - that's fair. Thanks!